crates/bevy_math/src/direction.rs

```diff
@@ -150,11 +150,39 @@ impl Dir2 {
         Self::new(Vec2::new(x, y))
     }
 
+    /// Creates a new [`Dir2`] from an angle in radians.
+    pub fn radians(angle: f32) -> Self {
+        let vec = Vec2::new(angle.cos(), angle.sin());
+
+        Dir2::new_unchecked(vec)
+    }
+
+    /// Creates a new [`Dir2`] from an angle in degrees.
+    pub fn degrees(angle: f32) -> Self {
+        Self::radians(angle.to_radians())
+    }
+
     /// Returns the inner [`Vec2`]
     pub const fn as_vec2(&self) -> Vec2 {
         self.0
     }
 
+    /// Validates that a direction is well-formed.
```

Comment:
Nitpick: Maybe the documentation could better explain that it does not check for the actual normalisation? I'm not super sure on this wording.